subsys/bluetooth/host/Kconfig

```
@@ -70,19 +70,48 @@ config BT_HCI_RESERVE
 	  Headroom that the driver needs for sending and receiving buffers. Add a
 	  new 'default' entry for each new driver.
 
-config BT_RECV_IS_RX_THREAD
-	# Hidden option set by the HCI driver to indicate that there's
-	# no need for the host to have its own RX thread.
-	# If this option has been enabled it is then the responsibility of the
-	# HCI driver to call bt_recv_prio from a higher priority context than
-	# bt_recv in order to avoid deadlocks.
-	# If this option is disabled then only bt_recv should be called.
-	bool
-	prompt "bt_recv is called from RX thread" if BT_NO_DRIVER
+
+choice BT_RECV_CONTEXT
+	prompt "BT RX Thread Selection"
+	default BT_RECV_DIRECT if BT_LL_SW_SPLIT || BT_H4
+	default BT_RECV_THREAD
+	help
+	  Select the Bluetooth RX Thread.
+
+config BT_RECV_DIRECT
+	bool "Direct call from hci driver thread"
+	help
+	  Indicate that there's no need for the host to have its own RX thread.
+	  If this option has been enabled it is then the responsibility of the
+	  HCI driver to call bt_recv_prio from a higher priority context than
+	  bt_recv in order to avoid deadlocks.
+	  If this option is disabled then only bt_recv should be called.
+
+config BT_RECV_THREAD
+	bool "Separate thread used"
+	help
+	  Indicate that host will create another thread, only used as BT RX.
+
+config BT_RECV_SYS_WORKQ
+	bool "System work queue used"
+	help
+	  System work queue used as bluetooth receive thread.
+
+	  Note: Refer to BT_RX_STACK_SIZE for the recommended minimum
+	  stack space of the system work queue.
+
+config BT_RECV_USR_WORKQ
+	bool "User work queue used"
+	help
+	  User work queue used as bluetooth receive thread.
+
+	  Note: Refer to BT_RX_STACK_SIZE for the recommended minimum
+	  stack space of the system work queue.
+endchoice
 
 config BT_RX_STACK_SIZE
 	int "Size of the receiving thread stack"
-	depends on BT_HCI_HOST || BT_RECV_IS_RX_THREAD
+	depends on BT_HCI_HOST || BT_RECV_DIRECT
```

I maybe be misunderstanding something here. 
Why is it not 
```suggestion
	depends on BT_HCI_HOST || BT_RECV_THREAD
```
?

I thought this is the stack size to be used when there is a thread in the host. Similar for BT_RX_PRIO